api/v1beta2/elb.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1beta2
+
+import (
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func (e *ELBHealthCheckTarget) ValidateELBHealthCheck() []*field.Error {
+	if e == nil {
+		return nil
+	}
+	var errs field.ErrorList
+	re := regexp.MustCompile(`^(TCP|SSL|HTTP|HTTPS|TLS|UDP):(\d+)\/?(\w*)$`)
```

Comment:
Build this outside so it's not rebuilt on each Validate call.

Reply:
Good idea!